src/app/services/courses-store.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, combineLatest, finalize, map, Observable, tap } from 'rxjs';
+import { Author } from '../app-model';
+import { Course } from '../features/courses/courses-model';
+import { AuthorsService } from './authors.service';
+
+import { CoursesService, SearchQueriesCourse } from './courses.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CoursesStoreService {
+  private courses$$ = new BehaviorSubject<Course[]>([]);
+  readonly courses$: Observable<Course[]> = this.courses$$.asObservable();
+
+  isLoading$ = new BehaviorSubject<boolean>(false);
+
+  constructor(private _coursesService: CoursesService, private _authorService: AuthorsService) {}
+
+  getAll(): Observable<Course[]> {
+    this.isLoading$.next(true);
+    return combineLatest([this._coursesService.getAll(), this._authorService.getAll()])
+      .pipe(map(([courses, authors]) => CoursesStoreService.replaceAuthorIdsToNamesInCourses(courses, authors)))
+      .pipe(
+        finalize(() => this.isLoading$.next(false)),
+        tap(courses => this.courses$$.next(courses))
+      );
+  }
+
+  createCourse(course: Course): Observable<any> {
+    this.isLoading$.next(true);
+    return this._coursesService.createCourse(course);
+  }
+
+  editCourse(course: Course): Observable<any> {
+    this.isLoading$.next(true);
+    return this._coursesService.editCourse(course);
+  }
+
+  filterCourse(queries: SearchQueriesCourse): Observable<any> {
+    this.isLoading$.next(true);
+    return combineLatest([this._coursesService.filterCourse(queries), this._authorService.getAll()])
+      .pipe(map(([courses, authors]) => CoursesStoreService.replaceAuthorIdsToNamesInCourses(courses, authors)))
```

Comment:
Why you use `pipe()` 3 times, you can chain all of the operators via comma.

Reply:
fixed